### PR TITLE
Update kopia sdk to upstream v0.14.1

### PR DESCRIPTION
**Pull request by leuyentran**
## Change Overview

In order to maintainin commit-parity for kopia between sdk in go.mo and kanister-tools image, these two PRs are a pair

* => https://github.com/kanisterio/kanister/pull/2416
* https://github.com/kanisterio/kanister/pull/2417

This PR:
- Upgrade kopia sdk to v0.14.1
- Fix breaking changes in `CacheOptions`

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -69,10 +69,10 @@ require (
...
 	cloud.google.com/go/compute/metadata v0.2.3 // indirect
 	cloud.google.com/go/iam v1.1.2 // indirect
-	cloud.google.com/go/storage v1.30.1 // indirect
-	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.3.0 // indirect
-	github.com/Azure/azure-sdk-for-go/sdk/internal v1.1.1 // indirect
-	github.com/Azure/azure-sdk-for-go/sdk/storage/azblob v1.0.0 // indirect; pinned - current kopia build referenced here requires v0.3.0
+	cloud.google.com/go/storage v1.32.0 // indirect
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.7.1 // indirect
+	github.com/Azure/azure-sdk-for-go/sdk/internal v1.3.0 // indirect
+	github.com/Azure/azure-sdk-for-go/sdk/storage/azblob v1.1.0 // indirect; pinned - current kopia build referenced here requires v0.3.0
```
Comment (KastenMike) [inline go.mod:75]: this pinned comment can be removed

File: pkg/kopia/repository/client.go [hunk 2]
```diff
@@ -62,10 +62,10 @@ func ConnectToAPIServer(
 
 	opts := &repo.ConnectOptions{
 		CachingOptions: content.CachingOptions{
-			CacheDirectory:            kopia.DefaultClientCacheDirectory,
-			MaxCacheSizeBytes:         int64(contentCacheMB << 20),
-			MaxMetadataCacheSizeBytes: int64(metadataCacheMB << 20),
-			MaxListCacheDuration:      content.DurationSeconds(defaultConnectMaxListCacheDuration.Seconds()),
+			CacheDirectory:         kopia.DefaultClientCacheDirectory,
+			ContentCacheSizeBytes:  int64(contentCacheMB << 20),
```
Comment (ankitjain235) [inline pkg/kopia/repository/client.go:66]: I see different flags(`ContentCacheSizeLimitBytes`, `MetadataCacheSizeLimitBytes`) being used in PR by @kale-amruta : https://github.com/kanisterio/kanister/pull/2393/files#diff-d96603cf822203f32cb69e1031279e1c357193ab6a01bbcf59ce08caa3dc3f8dR66
Just added comment at both of the PRs to confirm which of the flags to be used.

**Review by ankitjain235 — CHANGES_REQUESTED** [review]
Just blocking merge of the PR until the confusion in flag names is cleared.

File: pkg/kopia/repository/client.go (hunk 2 above)
Comment (kale-amruta) [reply pkg/kopia/repository/client.go:66]: https://github.com/kanisterio/kanister/pull/2393#discussion_r1369683114

**Review by ankitjain235 — APPROVED** [review]
LGTM

**Pull request merged**